addons/sale_project/models/sale_order_line.py

```diff
@@ -137,12 +137,24 @@ def _compute_qty_delivered(self):
     def _compute_analytic_distribution(self):
         super()._compute_analytic_distribution()
         for line in self:
-            if line.display_type or line.analytic_distribution or not line.product_id:
-                continue
             project = line.product_id.project_id or line.order_id.project_id
-            distribution = project._get_analytic_distribution()
-            if distribution:
-                line.analytic_distribution = distribution
+            if line.display_type or not line.product_id or not project:
+                continue
+
+            if line.analytic_distribution:
```

Comment:
Shouldn't the accounts from the project be more important than the accounts computed from the distribution models?

I summon thee @xavierbol !

Reply:
Coucou @william-andre 👋🏼 

Thanks for the ping! 🙂 

I think if the user manually set analytic distribution, we should keep his choice but if there is no analytic distribution set on the SOL then by default we will first take the distribution set on project. 🙂 

So, it seems to me the changes respect that, if I am not wrong. 🙂

Reply:
> if the user manually set analytic distribution

You mean when creating the line, or by configuring a distribution model? (= default value)

Reply:
when we create a line or when we set a project on the SO and the analytic distribution on the lines are still empty. 🙂

Reply:
Hmm, I don't really understand.
Could you give us the steps to create the tests needed according to you?

Reply:
Oh sorry @william-andre I misunderstood your question, actually.

No, we should first compute the analytic distribution from distribution models and then project accounts. Because if the user explicitly setup his analytic distributions, it would be weird for him to see the analytic distribution on the SOL are not like he configured his distribution models. 🙂 

My bad! 🙈